docs/tools/testing/testing_icetea.md

```diff
@@ -0,0 +1,95 @@
+
+##### Test discovery
+
+When using Icetea under the `mbed-os repository`, the `TEST_APPS folder` stores the CLI applications and the test cases. The `TEST_APPS` folder is located in the root of the `mbed-os` repository. If you are using Icetea somewhere else, you need to provide Icetea the information where the tests reside. You can do this with the `--tcdir <directory>` option.
+
+##### Test names
+
+Test names are defined in the Python scripts themselves. For example, the `SOCKET_BIND_PORT.py` contains two test cases; `TCPSOCKET_BIND_PORT` and `UDPSOCKET_BIND_PORT` are both defined in the Python file.
+
+#### Building tests
+
+You can build tests through Arm Mbed CLI. For information on using Mbed CLI, please see the [CLI documentation](/docs/development/tools/arm-mbed-cli.html).
+
+When you build tests for a target and a toolchain, the script first discovers the available tests, then the C++ applications linked to the tests and then builds the C++ applications in parallel.
+
+##### Building process
+
+The `test.py` script (not to be confused with `tests.py`) located under the `tools` directory handles the process for building tests. This handles the discovery and building of all test cases for a target and toolchain.
+
+The full build process is:
+
+1. Build the nontest code (all code not under a `TESTS` folder), but do not link it. The resulting object files are placed in the build directory.
+1. Find all tests that match the given target and toolchain.
+1. Find all the C++ applications that are defined in the tests.
+1. For each discovered C++ application, build all of its source files and link it with the nontest code that was built in step 1.
+1. If specified, create a test specification file and place it in the given directory for use by testing tools. This is placed in the build directory by default when using Mbed CLI.
+
+#### Running tests
+
+You can run automated tests through Mbed CLI with the `--icetea` option. You can find more information about this in the [Mbed CLI test documentation] (/docs/tools/offline/cli-test-debug.md).
+
+The testing process requires tests to be built and that a test specification JSON file exist that describes these available tests. See the [test specification format](https://github.com/ARMmbed/greentea#test-specification-json-formatted-input).
+
+The test specification JSON is similar for the Greentea and Icetea tests.
+
+The Icetea tool handles the actual testing process. To read more about this tool, please visit its [GitHub repository](https://github.com/ARMmbed/icetea).
+
+#### Writing tests
+
+##### Writing test cases
+
+For writing your own Python test scripts, you can read further information from [Icetea Test Case API](https://github.com/ARMmbed/icetea/blob/master/doc/tc_api.md).
```

Comment:
Note: We try not to link to GitHub. Is there another link we can use, instead? Maybe Doxygen?